### PR TITLE
making userName_ an AbstractButton should play nicely with ListView

### PR DESCRIPTION
**Pull request by maltee1**
Let's see if this works. If so, we can use it for Avatar, Reply and ImageMessage (did I forget any)?
Everything that involves more complex interaction than single and double tap/click (such as long press and right click) would probably need some customization and I don't know if that's worth it.
It also simplifies the code, because tap and hover are built into Control.
For some reason, moving the id to the AbstractButton messes with the cursorShape. This works, although I feel the id should be one level up. Maybe it's due to the way contentItem does things?

### CODE REVIEW

File: resources/qml/Avatar.qml [hunk 1]
```diff
@@ -18,13 +18,14 @@ Rectangle {
     property string displayName
     property alias textColor: label.color
     property bool crop: true
-
-    signal clicked(var mouse)
+    property color bgColor: Nheko.colors.alternateBase
```
Comment (redsky17) [inline resources/qml/Avatar.qml:21]: You can make this an alias instead, but would have to add and id to the background rectangle:

```
property alias bgColor: background.color

// ....
background: Rectangle: {
    id: background
    radius: Settings.avatarCircles  ? height / 2 : height / 8
}

Comment (maltee1) [reply resources/qml/Avatar.qml:21]: just for my understanding, why is an alias better? Fewer actual assignments?

**Comment by maltee1** [comment]
So this works fairly well.
I haven't changed ImageMessage.qml because grabbing images works (they're usually large enough)
I also haven't changed anything that uses long press or right click. It would be more complicated to get it to work properly and it doesn't bother me enough to be wanting to invest the time, since Qt6 should solve this anyway.
The TapHandlers for TimelineRow should now be the only ones that still cause problems, but only on empty areas on the bubble somehow.

File: resources/qml/Avatar.qml (hunk 1 above)
Comment (redsky17) [reply resources/qml/Avatar.qml:21]: Make sure to take a look at my edit... I messed around with formatting a few times there.  Also, to best answer your question, I'll refer you to the docs: https://doc.qt.io/qt-5/qtqml-syntax-objectattributes.html#property-aliases

Comment (maltee1) [reply resources/qml/Avatar.qml]: I see. That's better.

**Comment by deepbluev7** [comment]
Seems to work nicely and even simplify things, great job!

**Pull request merged**